modular_skyrat/modules/ashwalkers/code/buildings/railroad.dm

```diff
@@ -0,0 +1,127 @@
+	. = ..()
+	for(var/obj/structure/railroad/rail in range(1))
+		rail.update_appearance()
+
+/obj/structure/railroad/Destroy()
+	. = ..()
+	for(var/obj/structure/railroad/rail in range(1))
+		if(rail == src)
+			continue
+		addtimer(CALLBACK(rail, .proc/update_connection), 5)
+
+/obj/structure/railroad/update_appearance(updates)
+	icon_state = "rail"
+	var/turf/src_turf = get_turf(src)
+	for(var/direction in GLOB.cardinals)
+		var/obj/structure/railroad/locate_rail = locate() in get_step(src_turf, direction)
+		if(!locate_rail)
+			continue
+		icon_state = "[icon_state][direction]"
+	return ..()
+
+/obj/structure/railroad/attackby(obj/item/attacking_item, mob/user, params)
+	if(attacking_item.tool_behaviour == TOOL_CROWBAR)
+		if(!attacking_item.use_tool(src, user, 40, volume=75))
+			return
+		new /obj/item/stack/rail_track(get_turf(src))
+		qdel(src)
+		return
+	return ..()
+
+/obj/vehicle/ridden/rail_cart
+	name = "rail cart"
+	desc = "A wonderful form of locomotion. It will only ride while on tracks."
+	icon = 'modular_skyrat/modules/ashwalkers/icons/railroad.dmi'
+	icon_state = "railcart"
+
+/obj/vehicle/ridden/rail_cart/examine(mob/user)
+	. = ..()
+	. += span_notice("<br>Alt-Click to attach this rail cart to another.")
+	. += span_notice("<br>Click to access the inventory.")
+
+/obj/vehicle/ridden/rail_cart/Initialize(mapload)
+	. = ..()
+	attach_trailer()
+	AddElement(/datum/element/ridable, /datum/component/riding/vehicle/rail_cart)
+	AddComponent(/datum/component/storage/concrete)
+	var/datum/component/storage/added_storage = GetComponent(/datum/component/storage)
+	added_storage.max_combined_w_class = 21
+	added_storage.max_w_class = WEIGHT_CLASS_NORMAL
+	added_storage.max_items = 21
+
+/obj/vehicle/ridden/rail_cart/relaymove(mob/living/user, direction)
+	if(!canmove)
+		return FALSE
+	var/obj/structure/railroad/locate_rail = locate() in get_step(src, direction)
+	if(!locate_rail)
+		return FALSE
+	if(is_driver(user))
+		return relaydrive(user, direction)
+	return FALSE
+
+/obj/vehicle/ridden/rail_cart/AltClick(mob/user)
+	. = ..()
+	attach_trailer()
+
+/obj/vehicle/ridden/rail_cart/attack_hand(mob/living/user, list/modifiers)
+	. = ..()
+	var/datum/component/storage/rail_storage = GetComponent(/datum/component/storage)
+	rail_storage.open_storage(user)
+
+/obj/vehicle/ridden/rail_cart/proc/attach_trailer()
+	for(var/direction in GLOB.cardinals)
+		var/obj/vehicle/ridden/rail_cart/locate_cart = locate() in get_step(src, direction)
+		if(!locate_cart)
+			continue
+		if(locate_cart.trailer)
+			continue
+		locate_cart.trailer = src
+		break
+
+/datum/component/riding/vehicle/rail_cart
+	vehicle_move_delay = 0.5
+	ride_check_flags = RIDER_NEEDS_LEGS | RIDER_NEEDS_ARMS | UNBUCKLE_DISABLED_RIDER
+
+/datum/component/riding/vehicle/rail_cart/handle_specials()
+	. = ..()
+	set_riding_offsets(RIDING_OFFSET_ALL, list(TEXT_NORTH = list(0, 16), TEXT_SOUTH = list(0, 16), TEXT_EAST = list(0, 16), TEXT_WEST = list( 0, 16)))
```

Comment:
That list could also be made a define since it's the same for all the directions.
```suggestion
	set_riding_offsets(RIDING_OFFSET_ALL, list(TEXT_NORTH = list(0, 16), TEXT_SOUTH = list(0, 16), TEXT_EAST = list(0, 16), TEXT_WEST = list(0, 16)))
```

Reply:
It probably could be a define, but no other ones are.